src/main/java/org/minions/devfund/bruno/html/HtmlTag.java

```diff
@@ -0,0 +1,148 @@
+package org.minions.devfund.bruno.html;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * Represents information about a single HTML tag.
+ */
+public class HtmlTag {
+
+    protected final String element;
+    protected final boolean openTag;
+
+    /**
+     * Initializes an instance of {@link HtmlTag}.
+     *
+     * @param element   element name.
+     * @param isOpenTag
+     */
+    public HtmlTag(final String element, final boolean isOpenTag) {
+        this.element = element.toLowerCase();
+        openTag = isOpenTag;
+    }
+
+    /**
+     * Gets the element name (String) specified in this tag.
+     *
+     * @return the element name.
+     */
+    public String getElement() {
+        return element;
+    }
+
+    /**
+     * Checks whether this is the opening tag.
+     *
+     * @return false if is open the tag.
+     */
+    public boolean isOpenTag() {
+        return openTag && !isSelfClosing();
+    }
+
+    /**
+     * Checks whether an HtmlTag other is the matching open/close tag to itself.
+     *
+     * @param other Html tag
+     * @return
+     */
+    public boolean matches(final HtmlTag other) {
+        return other != null
+                && element.equalsIgnoreCase(other.element)
+                && openTag != other.openTag;
+    }
+
+    /**
+     * Checks whether a tag is self-closing.
+     *
+     * @return if is self closing return true.
+     */
+    public boolean isSelfClosing() {
+        return SELF_CLOSING_TAGS.contains(element);
+    }
+
+    public boolean equals(Object obj) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This class overrides "equals()" and should therefore also override "hashCode()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1206)